stile/sys_tests.py

```diff
@@ -859,12 +894,20 @@ class WhiskerPlotStarSysTest(WhiskerPlotSysTest):
 
     def __call__(self, array, linewidth = 0.01, scale = None, figsize = None,
                  xlim = None, ylim = None):
+        if 'CCD' in array.dtype.names:
+            fields = list(self.required_quantities[0]) + ['CCD']
+        else:
+            fields = list(self.required_quantities[0])
+        self.data = numpy.rec.fromarrays([array[field] for field in fields], names = fields)
         return self.whiskerPlot(array['x'], array['y'], array['g1'], array['g2'], array['sigma'],
                                 linewidth = linewidth, scale = scale, figsize = figsize,
                                 xlabel = r'$x$ [pixel]', ylabel = r'$y$ [pixel]',
                                 size_label = r'$\sigma$ [pixel]',
                                 xlim = xlim, ylim = ylim, equal_axis = True)
 
+    def getData(self):
```

Comment:
You could make this function a function of `WhiskerPlotSysTest` and then it wouldn't need to be repeated in all the subclasses